src/lib/io/sol-uart-impl-riot.c

```diff
@@ -119,48 +150,99 @@ sol_uart_open(const char *port_name, const struct sol_uart_config *config)
     uart_poweron(uart->id);
     ret = sol_interrupt_scheduler_uart_init_int(uart->id,
         baud_rata_table[config->baud_rate], uart_rx_cb, uart,
-        &uart->async.handler);
+        &uart->handler);
     SOL_INT_CHECK_GOTO(ret, != 0, fail);
 
-    uart->async.rx_cb = config->rx_cb;
-    uart->async.rx_user_data = config->rx_cb_user_data;
-    uart->async.tx_buffer = NULL;
+    uart->rx_cb = config->rx_cb;
+    uart->tx_cb = config->tx_cb;
+    uart->user_data = config->user_data;
+    sol_ptr_vector_init(&uart->pending_blobs);
+    sol_buffer_init_flags(&uart->rx_buffer, NULL, 0,
+        SOL_BUFFER_FLAGS_DEFAULT | SOL_BUFFER_FLAGS_NO_NUL_BYTE);
```

Comment:
since you do fixed-size buffer in a much more capable system (linux), why not do the same here? As I said before, we keep using sol_buffer and eventually, if there is need, we can change the buffer to fixed capacity or not based on kconfig.